homeassistant/components/squeezebox/media_player.py

```diff
@@ -294,6 +308,26 @@ async def async_update(self) -> None:
                     self.hass, SIGNAL_PLAYER_REDISCOVERED, self.rediscovered
                 )
 
+    async def _async_sync_favorites(self, loop):
+        _command = ["favorites"]
+        _command.extend(["items","0","100"])
+        _res = (await self._player.async_query(*_command))["loop_loop"]
+        _favs = []
+        for _fav in _res:
+            _favs+=[_fav["name"]]
+        self._sources = tuple(_favs)
```

Comment:
Favorites should be exposed as media sources and not as sources. Media sources are fetched on demand, so the sync service should be removed.

Reply:
Thanks @balloob .  That'll take changes to the python library as the browse methods don't support favorites at the moment and this libs don't seem to have been touched in quite a few years.  @rajlaud, are you open to a PR on the python library if I can figure out the changes to add favorites as a category?  PS, I quite liked the favourites being on the main UI of the cards as I'd seen on a couple of other (custom) components, but understand they should really be in the media browse.

Reply:
You can "fake" a favorites folder in the media browsing inside Home Assistant and just list the same content as you do here. That's also how we do it for some other integrations 👍

Reply:
Ah c'mon @balloob , you can't tease us like that.  Any pointers at which others might do it this way - seems much more sensible than burying a whole separate path in the browse.

Reply:
Sonos adds [an entry for favorites](https://github.com/home-assistant/core/blob/dev/homeassistant/components/sonos/media_browser.py#L299) in their root payload. And then when a user browses it, it will list data from [a specific endpoint](https://github.com/home-assistant/core/blob/dev/homeassistant/components/sonos/media_browser.py#L157)

Reply:
@peteS-UK hows this going?

Reply:
Hi @pssc I haven't had a chance to look at this lately - my laptop died and my old laptop isn't upto running a dev container so I've been without a dev environment.  I'll try and revisit it (although I still use the sources option at present as they're then on the main UI of the media player component, but I do understand the desire for consistency as well).

Reply:
Hi @pssc OK - I've made the changes to add a basic favorites category to the media browser - it doesn't support folders as yet, but I don't think they're that widely used in LMS world - I'll look at that for a v2 - messy bit is there isn't a call to say "what favs are in this folder".

The only issue I have atm is that the media browser doesn't display the favorites in grid view, but works fine in list view - which makes no sense to me.  I've posted a question in discord for now.  Once I have an answer, I'll likely close this PR and raise a new one since the baseline has moved such a lot since this original PR.

Reply:
@peteS-UK happy to help if I can.  I too would like to see this available via HA to help I was looking to add Favs to the underlying browse code in the supporting library, think I got as far as the unit tests.

Reply:
Attached are the 2 files - very minor change to media_player - just on the playing - then modifications to build_item_response and library_payload in browse_media.
[new.zip](https://github.com/user-attachments/files/16603072/new.zip)
Really odd that it actually works fine in list view and I've got no idea where the media browser window code lives so don't know where to look.

Reply:
https://discord.com/channels/330944238910963714/1272937051922108498

Reply:
Once I can figure out the problem, I'll submit a new PR, or you could make the changes in the lib as well

Reply:
@pssc If you get a chance to try these changes out, I'd be very interested to know if you see the same issue with Grid but not List view - if so, I might submit an issue around it as it doesn't really make sense that the same code would work for one view but not another.

Reply:
Ok.  I found the problem.  media_class has to be one of the predefined values.  No idea why list view works but grid doesn't but hey ho.  

I'll tidy it up later and post a new PR.

Reply:
@peteS-UK sure I can pull it down and give it a test

Reply:
@pssc  - take a look at https://github.com/home-assistant/core/pull/123953